sdk/src/rpc.rs

```diff
@@ -1163,12 +1163,33 @@ pub async fn format_denominated_amount(
 /// Look up the IBC denomination from a IbcToken.
 pub async fn query_ibc_denom<N: Namada>(
     context: &N,
-    token: &Address,
+    token: impl AsRef<str>,
     owner: Option<&Address>,
 ) -> String {
-    let hash = match token {
-        Address::Internal(InternalAddress::IbcToken(hash)) => hash.to_string(),
-        _ => return token.to_string(),
+    let hash = match Address::decode(token.as_ref()) {
+        Ok(Address::Internal(InternalAddress::IbcToken(hash))) => {
+            hash.to_string()
+        }
+        Ok(_) => return token.as_ref().to_string(),
+        Err(_) => match namada_core::types::ibc::is_ibc_denom(token.as_ref()) {
+            Some((trace_path, base_denom)) => {
+                let base_token = context
+                    .wallet()
+                    .await
+                    .find_address(&base_denom)
+                    .map(|addr| addr.to_string())
+                    .unwrap_or(base_denom);
+                return format!("{trace_path}/{base_token}");
+            }
+            None => {
+                return context
+                    .wallet()
+                    .await
+                    .find_address(token.as_ref())
+                    .map(|addr| addr.to_string())
+                    .unwrap_or(token.as_ref().to_string());
+            }
```

Comment:
is this the correct behavior? users could have populated their wallet with an alias that might coincide with some other chain's address format. it's odd, but certainly possible. perhaps we should document in the cli the order of parsing/look-ups

Reply:
Good point. We shouldn't look up the token address with the wallet of the destination chain.

When `ibc-gen-shielded` is requested, a user is trying to send the token of another chain. A user should be able to get the raw address of the token on the source chain.

And, if the token address is invalid, though the proof will be generated, the following ibc-transfer will fail.

(When `query_ibc_denom` is called from `ibc-transfer`, the token should be parsed and this `Err` branch shouldn't be reached.)